functions/src/update-stats.ts

```diff
@@ -21,39 +21,40 @@ export const updateStats = functions.firestore
     }
     const vote = voteSnapshot.data()! as Vote;
     const voteId: string = voteSnapshot.id;
-    const statsCollection = firebase.firestore().collection("stats");
-
-    await firebase.firestore().runTransaction(async transaction => {
-      const doc = await transaction.get(statsCollection.doc(vote.campaign));
-      const storedVote = await transaction.get(
-        statsCollection
-          .doc(vote.campaign)
-          .collection("votes")
-          .doc(voteId)
+    updateStatsFunction(vote, voteId);
+  });
+
+export const updateStatsFunction = async (vote: Vote, voteId: string) => {
+  const statsCollection = firebase.firestore().collection("stats");
+  await firebase.firestore().runTransaction(async transaction => {
+    const doc = await transaction.get(statsCollection.doc(vote.campaign));
+    const storedVote = await transaction.get(
+      statsCollection
+        .doc(vote.campaign)
+        .collection("votes")
+        .doc(voteId)
+    );
+
+    if (storedVote.exists) {
+      console.info(
+        "This vote(" + voteId + ") has already been counted, aborting;"
```

Comment:
A template string might be more appropriate here. Maybe you could check if a tslint rule exists to enforce their use.